src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -657,6 +657,15 @@ void GcsActorManager::ReconstructActor(const ActorID &actor_id, bool need_resche
         }));
     gcs_actor_scheduler_->Schedule(actor);
   } else {
+    // For detached actors, make sure to clean up their names.
+    // This doesn't have to be called after storing actor state dead because
+    // GCS is reliable and we can guarantee dead states will be persisted.
+    if (actor->IsDetached()) {
+      auto it = named_actors_.find(actor->GetName());
+      if (it != named_actors_.end() && it->second == actor->GetActorID()) {
+        named_actors_.erase(actor->GetName());
```

Comment:
```suggestion
        named_actors_.erase(it);
```

(avoids another lookup)

Reply:
It's okay! Not a big deal, just good to get in the habit of this code style :)